builder/builder-next/executor_nolinux.go

```diff
@@ -0,0 +1,34 @@
+//go:build !linux
+
+package buildkit
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/docker/docker/daemon/config"
+	"github.com/docker/docker/libnetwork"
+	"github.com/docker/docker/pkg/idtools"
+	"github.com/moby/buildkit/executor"
+	"github.com/moby/buildkit/executor/oci"
+	resourcetypes "github.com/moby/buildkit/executor/resources/types"
+)
+
+func newExecutor(_, _ string, _ *libnetwork.Controller, _ *oci.DNSConfig, _ bool, _ idtools.IdentityMapping, _ string) (executor.Executor, error) {
+	return &stubExecutor{}, nil
+}
+
+type stubExecutor struct{}
+
+func (w *stubExecutor) Run(ctx context.Context, id string, root executor.Mount, mounts []executor.Mount, process executor.ProcessInfo, started chan<- struct{}) (resourcetypes.Recorder, error) {
+	return nil, fmt.Errorf("buildkit executor not implemented for %s", runtime.GOOS)
```

Comment:
Nit: build the string at compile time as `runtime.GOOS` is a compile-time constant.
```suggestion
	return nil, errors.New("buildkit executor not implemented for "+runtime.GOOS)
```